### DISCUSSION TITLE
Model Animation with Posenet

### DISCUSSION THREAD

**Discussion by Kashif-E** [sceneview/sceneform-android#105] [Q&A]
I am trying to move a 3d model by accessing its bones and moving it to the point that a posenet model returns. Is it possible? I have been looking into it but have not found anything related to this, it would be really helpful if you can give me some guidance

**Comment by ThomasGorisse** [comment]
You can modify the model bones matrice like this:
```
TransformableNode model = new TransformableNode(arFragment.getTransformationSystem());
model.setParent(anchorNode);
RenderableInstance modelInstance = modelNode.setRenderable(this.model);
RenderableManager renderableManager = EngineInstance.getEngine().getRenderableManager();
int entityInstance = renderableManager.getInstance(modelInstance.getEntity());
renderableManager.setBonesAsMatrices(entityInstance, matrices, boneCount, offset)
```
You can ask your more 3D Engine specific questions on the [Filament Repo](https://github.com/google/filament)

**Comment by Kashif-E** [comment]
from where did you get the bone count, offset, and matrices?

**Reply by ThomasGorisse** [reply]
Ask it on the Filament repo!!!!
